### PR TITLE
feat: Added dark theme toggle [#15]

### LINKED ISSUE #15: Add Dark Theme toggle
I want to add a toggle to swap the website to a dark theme. There could also be an option to match the system theme too

### PR DESCRIPTION
**Pull request by yyjlincoln**
@Aniket762 and I have worked together on this PR. In this, we have added a toggle button that gives 3 options light mode, dark mode, and auto which displays according to the system preference. 

A glimpse of our work: [Demo](https://youtu.be/1xv1J13I970) Credit goes to @yyjlincoln for this! 

Features:
- The support of auto|light|dark
- Smooth transition between states
- No eye-blinding flash when switching pages
- When set to auto, it can listen to your phone's dark mode events and adjust its color scheme automatically
- Written in vanilla javascript, no additional libraries are used
- Flexible/modular design with functions such as `beforeModeChange` and `afterModeChange`, meaning that you can easily add more functionality to it.

- [X] My Pod Leader knows I'm working on this Pull Request
- [X] I've explained what the Pull Request is adding.
- [X] I've explained why this is important.

*This is an updated PR as all commits have now migrated to the main repository - allows better collaboration as everyone now has the access to the code and makes things less complicated. [#22]

This pull request closes #15.

### CODE REVIEW

**Review by Aniket762 — COMMENTED** [review]
LGTM!

File: index.html [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
+
 <!DOCTYPE html>
 <html lang="en">
 {% include header.html %}
 
-<body>
-    {% include navbar.html %}
+<body class="hidden">
+  {% include navbar.html %} {% include profile.html %} {% include
+  description.html %} {% include projects.html %} {% include toggle.html %}
```
Comment (ditsuke) [inline index.html:10]:
```suggestion
    {% include navbar.html %}
  
    {% include profile.html %}
   
    {% include description.html %}
   
    {% include projects.html %}
   
    {% include toggle.html %}
```

Comment (yyjlincoln) [reply index.html]: Seems like an issue with Aniket's formatter! We really should try and convince everybody to use the `.editorconfig` file!
Thanks for the fix. That will make git diff way easier.

**Review by VaibhaveS — APPROVED** [review]
Perfect!

**Comment by Aniket762** [comment]
@gmelodie this PR is approved by 2 reviewers I guess we are good to merge it! 

@yyjlincoln do you second this?

**Comment by yyjlincoln** [comment]
> @gmelodie this PR is approved by 2 reviewers I guess we are good to merge it! 
> 
> 
> 
> @yyjlincoln do you second this? 

I think it is ready to go but I think @gmelodie told us that he was the only person that's gonna merge anything into main.

**Review by ditsuke — APPROVED** [review]

File: _includes/header.html [hunk 2]
```diff
@@ -3,8 +3,7 @@
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
     <meta property="og:title" content="Personal Portfolio">
-    <meta property="og:description"
-        content="My Personal Portfolio">
+    <meta property="og:description" content="My Personal Portfolio">
```
Comment (gmelodie) [inline _includes/header.html:6]: Do not change things unrelated to the issue at hand (:scissors:)

File: _includes/header.html [hunk 3]
```diff
@@ -17,15 +16,19 @@
     <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/js/bootstrap.min.js"
         integrity="sha384-B0UglyR+jN6CkvvICOB2joaf5I4l3gm9GU6Hc1og6Ls7i6U/mkkaduKaBhlAXv9k"
         crossorigin="anonymous"></script>
-    <script src="https://jsxss.com/assets/xss.js"
-        crossorigin="anonymous"></script>
+    <script src="https://jsxss.com/assets/xss.js" crossorigin="anonymous"></script>
```
Comment (gmelodie) [inline _includes/header.html:19]: :scissors:

File: _includes/navbar.html [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
-                <img src="./assets/img/mlh-prep.png" />
-            </div>
-        </a>
-        
+  <div class="nav-content">
+    <a href="{{ '/' | relative_url }}">
+      <div class="nav-logo">
+        <img src="./assets/img/mlh-prep.png" />
+      </div>
+    </a>
```
Comment (gmelodie) [inline _includes/navbar.html:7]: :scissors:

File: _includes/toggle.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        bottom: 3rem;
+        right: 3rem;
+        border-radius: 1rem;
+        display: flex;
+        flex-direction: row;
+        user-select: none;
+        cursor: pointer;
+        font-weight: bold;
+    }
+</style>
```
Comment (gmelodie) [inline _includes/toggle.html:32]: Maybe put this in `page.scss` instead

File: _layouts/default.html [hunk 6]
```diff
@@ -1,11 +1,14 @@
 ---
 ---
+
```
Comment (gmelodie) [inline _layouts/default.html:3]: :scissors:

File: _layouts/page.html [hunk 7]
```diff
@@ -2,11 +2,12 @@
 layout: default
 ---
 
+
```
Comment (gmelodie) [inline _layouts/page.html:5]: :scissors:

File: _layouts/page.html [hunk 8]
```diff
@@ -2,11 +2,12 @@
...
 <div class="banner">
     <div class="container">
         <div class="logo">
             <a href="/">
-                <img src="/assets/img/logo.svg"/>
+                <img src="/assets/img/logo.svg" />
```
Comment (gmelodie) [inline _layouts/page.html:10]: :scissors:

File: _sass/all.scss [hunk 9]
```diff
@@ -26,65 +27,72 @@
...
+  margin-right: 20px;
+  padding: 10px;
 
-    img {
-        width: 100%;
-    }
+  img {
+    width: 100%;
+  }
 }
```
Comment (gmelodie) [inline _sass/all.scss:98]: :scissors:

File: _sass/navbar.scss [hunk 10]
```diff
@@ -1,36 +1,73 @@
...
-    width: 100%;
-    text-align: right;
-    padding-top: 0px;
-    padding-right: 0px;
+  display: block;
+  width: 100%;
+  text-align: right;
+  padding-top: 0px;
+  padding-right: 0px;
+}
```
Comment (gmelodie) [inline _sass/navbar.scss:36]: :scissors:

File: index.html [hunk 11]
```diff
@@ -1,17 +1,21 @@
...
+<body class="hidden">
     {% include navbar.html %}
-
+  
     {% include profile.html %}
-    
+   
     {% include description.html %}
-    
+
```
Comment (gmelodie) [inline index.html:14]: :scissors:

**Review by gmelodie — CHANGES_REQUESTED** [review]
Nice work! Looking really good :)

**Comment by yyjlincoln** [comment]
Merging to `main` as:
- there are at least 2 approvals.
- all checks have passed.
- fixed up the requested changes from @gmelodie (5b9ff6f). There is no change to functionality - just formatting.

**Comment by yyjlincoln** [comment]
- Undid the previous merge (`main -> this branch`) as it brought the bug (#30, introduced in f4100c7e73dc26eb6aa7623eea4df2db4dd5067d) to this branch.
- As the bug has been fixed and merged into `main` in #31, I merged `the updated main -> this branch` to resolve any outstanding conflicts.
- Now this branch is ready to be merged back to `main` as all conflicts have been fixed.

**Pull request merged**